### DISCUSSION TITLE
Possible regression in custom list handling?

### DISCUSSION THREAD

**Discussion by xvello** [uBlockOrigin/uBlock-issues#2659] [Q&A] [answered]
Hello! I'm the maintainer of letsblock.it, a project that allows its users to build a custom filter list and get it automatically updated on all their devices. Generated lists have the following header, to refresh as fast as uBO allows:

```
! Title: letsblock.it - My filters
! Expires: 12 hours
! Homepage: https://letsblock.it
! License: https://github.com/letsblockit/letsblockit/blob/main/LICENSE.txt
```

While looking at the usage metrics for the official instance, I noticed a sudden and steady drop in list downloads, that starts around April 28th. The metric below tracks the lists that have not been updated for 7+days, and it starts going up on May 5th, one week after:

<img width="1146" alt="Screenshot 2023-05-16 at 21 18 49" src="https://github.com/uBlockOrigin/uBlock-issues/assets/6241083/cc22f562-1d29-4842-8760-d57d2ac9e389">

That metric goes up when users delete the list from their config, or reset their browser and don't re-install the list. Some churn is expected, but this looks too sudden and linear to not be suspicious. The timing seems to match the rollout of the latest 1.49.2 release, which is why I'm wondering whether there could be a regression in that release, either:

- failing to refresh the custom lists, or refreshing them less often
- disabling / losing the custom lists due to a config migration

I ruled out a server issue by checking the server logs: there's no increase in 400s or 500s errors for the list download endpoint. AdGuard's [latest release](https://github.com/AdguardTeam/AdguardBrowserExtension/releases/tag/v4.1.55) was on Apr 6th, so the timing does not line up as nicely as uBO.

**Comment by gorhill** [comment, marked as answer]
Most likely related to <https://github.com/gorhill/uBlock/commit/1a9a8aae0c8969dad9743e31dd465c8892b89f32>.

What is the _y_ axis unit on your graph?

**Reply by xvello** [reply]
The y axis is the total number of users that have not refreshed their list in the last 7 days. We can see a slow but consitant increase for months (users churning out), and a big acceleration one week after the 1.49.2 release.

**Reply by xvello** [reply]
Good news to know that a fix is already underway. Could you please help me understand the user impact here?

- are rules from custom lists still active but not refreshed, or are users losing these rules?
- will 1.49.3 be able to recover the list, or will users need to re-add the custom list?

**Reply by gorhill** [reply]
Users will need to re-enable the lists, they are still imported, just deactivated. There is nothing else that can be done to fix this -- there is no way for uBO to know which lists were enabled among all the ones imported.

**Reply by xvello** [reply]
Thanks for your reply!
I just tested on Firefox, and clicking on the `abp:subscribe` link allows to re-enable the list, even if it's already installed. I was afraid that it wouldn't, and that more involved manipulations would be required.
I'll work on a communication plan towards affected users. Is it OK to run communications now, or shall I wait for the next release to rollout, to avoid users needing to re-enable twice?

**Reply by gorhill** [reply]
I am assuming majority of people already transitioned from 1.48.x to 1.49.x, I don't think waiting for next release will make a difference.

**Reply by xvello** [reply]
By looking at the issues, I'm getting the feeling that the bug only happened in Chrome, do you have enough info to confirm whether all uBO users have been affected, or only the ones using Chrome & derivatives?

**Reply by gorhill** [reply]
The code I fixed doesn't depend on browser vendor.
